pandas/core/indexes/datetimelike.py

```diff
@@ -124,8 +123,11 @@ def ceil(self, freq):
         return self._round(freq, np.ceil)
 
 
-class DatetimeIndexOpsMixin(object):
+class DatetimeIndexOpsMixin(DatetimeLikeArray):
     """ common ops mixin to support a unified interface datetimelike Index """
+    inferred_freq = cache_readonly(DatetimeLikeArray.inferred_freq.fget)
```

Comment:
Comment here to note why we do it like this (array is mutable, index is immutable).

Reply:
Will do.